nilearn/plotting/__init__.py

```diff
@@ -30,12 +30,12 @@ def _set_mpl_backend():
         if 'inline' in current_backend or 'nbagg' in current_backend:
             return
         # Set the backend to a non-interactive one for unices without X
-        if (os.name == 'posix' and 'DISPLAY' not in os.environ
-                and not (sys.platform == 'darwin'
-                         and 'macosx' in current_backend)
-        ):
+        # (see gh-2560)
+        if (sys.platform not in ('darwin', 'win32') and
+                'DISPLAY' not in os.environ):
```

Comment:
this changes a bit the logic as you assume that a mac will not be used as a remove headless server but i agree it's unlikely

I can live with that

Reply:
Related discussion is in #2560

FWIW I think/assume Travis and GitHub macOS CIs are headless and they gracefully fall back to software rendering anyway.